base/compiler/abstractinterpretation.jl

```diff
@@ -854,13 +878,27 @@ function abstract_call_method_with_const_args(interp::AbstractInterpreter, resul
     if !const_prop_enabled(interp, sv, match)
         return nothing
     end
-    val = concrete_eval_call(interp, f, result, arginfo, sv)
-    if val !== nothing
-        add_backedge!(val.const_result.mi, sv)
-        return val
+    res = concrete_eval_call(interp, f, result, arginfo, sv)
+    if isa(res, ConstCallResults)
+        add_backedge!(result.edge, sv)
```

Comment:
```suggestion
        add_backedge!(res.const_result.mi, sv)
```
This is more type-stable.